### PR TITLE
[Merged by Bors] - feat port: Data.PEquiv

### PR DESCRIPTION
**Pull request by ChrisHughes24**
ee0c179cd3c8a45aa5bffbf1b41d8dbede452865

### CODE REVIEW

File: Mathlib/Data/PEquiv.lean [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+theorem trans_refl (f : α ≃. β) : f.trans (PEquiv.refl β) = f := by
+  ext; dsimp [PEquiv.trans]; simp
+#align pequiv.trans_refl PEquiv.trans_refl
+
+protected theorem inj (f : α ≃. β) {a₁ a₂ : α} {b : β} (h₁ : b ∈ f a₁) (h₂ : b ∈ f a₂) : a₁ = a₂ :=
+  by rw [← mem_iff_mem] at *; cases h : f.symm b <;> simp_all
+#align pequiv.inj PEquiv.inj
+
+/-- If the domain of a `PEquiv` is `α` except a point, its forward direction is injective. -/
+theorem injective_of_forall_ne_is_some (f : α ≃. β) (a₂ : α)
```
Comment (semorrison) [inline Mathlib/Data/PEquiv.lean:185]:
```suggestion
theorem injective_of_forall_ne_isSome (f : α ≃. β) (a₂ : α)
```
and below.

File: Mathlib/Data/PEquiv.lean [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+
+theorem le_def {f g : α ≃. β} : f ≤ g ↔ ∀ (a : α) (b : β), b ∈ f a → b ∈ g a :=
+  Iff.rfl
+#align pequiv.le_def PEquiv.le_def
+
+instance : OrderBot (α ≃. β) :=
+  { instBotPEquiv with bot_le := fun _ _ _ h => (not_mem_none _ h).elim }
+
+instance [DecidableEq α] [DecidableEq β] : SemilatticeInf (α ≃. β) :=
+  { -- `split_ifs; finish` closes this goal from here
```
Comment (semorrison) [inline Mathlib/Data/PEquiv.lean:441]: This comment has mysteriously migrated: it's meant to be on the `split_ifs` below.

File: Mathlib/Data/PEquiv.lean [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+
+end Order
+
+end PEquiv
+
+namespace Equiv
+
+variable {α : Type _} {β : Type _} {γ : Type _}
+
+/-- Turns an `equiv` into a `PEquiv` of the whole type. -/
```
Comment (semorrison) [inline Mathlib/Data/PEquiv.lean:478]:
```suggestion
/-- Turns an `Equiv` into a `PEquiv` of the whole type. -/
```

**Comment by semorrison** [comment]
bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**